core/quantity/src/Formatter/Formatter.ts

```diff
@@ -244,7 +244,30 @@ export class Formatter {
       }
 
       if (i < (spec.format.units?.length ?? 0) - 1) {
-        const wholePart = Math.trunc(unitValue);
+        let wholePart = Math.trunc(unitValue);
+
+        // Check if the remaining fractional part will round up to a full unit in the next (smaller) component
+        if (spec.format.type === FormatType.Fractional && i === spec.unitConversions.length - 2) {
+          // For the second-to-last unit with fractional formatting, check if rounding causes carry-over
+          const fractionalPart = unitValue - wholePart;
+          const nextUnitValue = applyConversion(fractionalPart, spec.unitConversions[i + 1].conversion);
+
+          // Create a FractionalNumeric to determine what the rounded value would be
+          const fn = new FractionalNumeric(Math.abs(nextUnitValue), spec.format.precision as FractionalPrecision, true);
```

Comment:
This cast is jumping out. Why are you sure the precision is FractionalPrecision at this point?